File: editor/en/[1]Two Sum.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let map = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (!map.has(nums[i])) {
            map.set(target - nums[i], i);
        } else {
            return [map.get(nums[i]), i];
        }
    }
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1]Two Sum.go
//leetcode submit region begin(Prohibit modification and deletion)
func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		complement := target - num

		if index, found := numMap[complement]; found {
			return []int{index, i}
		}

		numMap[num] = i
	}

	return []int{}
}

//leetcode submit region end(Prohibit modification and deletion)
